src/lib/logical_query_plan/jit_aware_lqp_translator.cpp

```diff
@@ -0,0 +1,273 @@
+  const auto column_id = input_node->find_output_column_id(lqp_column_reference);
+  const auto stored_table_node = std::dynamic_pointer_cast<const StoredTableNode>(lqp_column_reference.original_node());
+  const auto projection_node = std::dynamic_pointer_cast<const ProjectionNode>(lqp_column_reference.original_node());
+
+  if (column_id && stored_table_node) {
+    // we reached a "raw" data column and register it as an input column
+    const auto original_column_id = lqp_column_reference.original_column_id();
+    const auto table_name = stored_table_node->table_name();
+    const auto table = StorageManager::get().get_table(table_name);
+    const auto data_type = table->column_data_type(original_column_id);
+    const auto is_nullable = table->column_is_nullable(original_column_id);
+    const auto tuple_value = jit_source.add_input_column(data_type, is_nullable, column_id.value());
+    return std::make_shared<JitExpression>(tuple_value);
+  } else if (projection_node) {
+    // if the LQPColumnReference references a computed column, we need to compute that expression as well
+    const auto lqp_expression = projection_node->column_expressions()[lqp_column_reference.original_column_id()];
+    return _translate_to_jit_expression(*lqp_expression, jit_source, input_node);
+  } else {
+    Fail("Could not get column from anywhere.");
+  }
+}
+
+std::shared_ptr<const JitExpression> JitAwareLQPTranslator::_translate_to_jit_expression(
+    const AllParameterVariant& value, JitReadTuple& jit_source,
+    const std::shared_ptr<AbstractLQPNode>& input_node) const {
+  if (is_lqp_column_reference(value)) {
+    return _translate_to_jit_expression(boost::get<LQPColumnReference>(value), jit_source, input_node);
+  } else if (is_variant(value)) {
+    const auto variant = boost::get<AllTypeVariant>(value);
+    const auto tuple_value = jit_source.add_literal_value(variant);
+    return std::make_shared<JitExpression>(tuple_value);
+  } else {
+    Fail("Unexpected parameter type.");
+  }
+}
+
+bool JitAwareLQPTranslator::_has_another_condition(const std::shared_ptr<AbstractLQPNode>& node) const {
+  auto current_node = node;
+  while (current_node->type() == LQPNodeType::Projection) {
+    current_node = current_node->left_input();
+  }
+  return current_node->type() == LQPNodeType::Predicate || current_node->type() == LQPNodeType::Union;
+}
+
+bool JitAwareLQPTranslator::_node_is_jitable(const std::shared_ptr<AbstractLQPNode>& node) const {
+  switch (node->type()) {
+    case LQPNodeType::Predicate:
+      return std::dynamic_pointer_cast<PredicateNode>(node)->scan_type() == ScanType::TableScan &&
+             std::dynamic_pointer_cast<PredicateNode>(node)->predicate_condition() != PredicateCondition::Between;
+    case LQPNodeType::Projection:
+    case LQPNodeType::Union:
+      return true;
+    default:
+      return false;
+  }
+}
+
+void JitAwareLQPTranslator::_breadth_first_search(
```

Comment:
A search method that doesn't give you a result? If `func` is expected to keep track of the results, it is not a search, maybe an `apply`.

Reply:
IMO `apply` does not really capture that multiple elements might be affected. I went with `visit` for now. Ok?